app/src/androidTest/java/com/example/sharingang/MapFragmentTest.kt

```diff
@@ -33,19 +33,41 @@ class MapFragmentTest {
     val grantPermissionRule: GrantPermissionRule =
         GrantPermissionRule.grant(Manifest.permission.ACCESS_FINE_LOCATION)
 
+
+    private val itemTitle = "T"
+
     @Test
     fun itemsAddedAreDisplayedOnTheMap() {
         navigate_to(R.id.newItemFragment)
+        onView(withId(R.id.editItemTitle)).perform(
+            ViewActions.typeText(itemTitle),
+            ViewActions.closeSoftKeyboard()
+        )
         onView(withId(R.id.new_item_get_location)).perform(click())
-        Thread.sleep(3000)
+        Thread.sleep(6000)
         onView(withId(R.id.createItemButton)).perform(click())
         navigate_to(R.id.mapFragment)
-        val text = onView(withId(R.id.location_display))
-        //text.check(matches(withText("")))
         Thread.sleep(6000)
-        text.check(matches(withText(containsString("Your location"))))
+        var activity: Activity? = null
+        activityRule.scenario.onActivity {
+            activity = it
+        }
+        var result = 0
+        val resourceId: Int =
+            activity!!.resources.getIdentifier("status_bar_height", "dimen", "android")
+        if (resourceId > 0) {
+            result = activity!!.resources.getDimensionPixelSize(resourceId)
+        }
         val device = UiDevice.getInstance(getInstrumentation())
-        val marker = device.findObject(UiSelector().descriptionContains(""))
-        marker.click()
+        device.click(device.displayWidth / 2, device.displayHeight / 2 + result)
```

Comment:
This seems like an error prone method, but I guess there's not really any alternative to working with the map is there?

If this causes even flakier CI, we're going to have fun later.

Reply:
Unfortunately there is no way to directly click on a specific marker on the map. Instead, I decided to place an object at "my" location, then wait for the map to center itself around "my" location, and then I click at the center of the screen, taking into account the status bar width.